examples/dfp_axis/src/main/java/dfp/axis/v201708/creativeservice/CreateCustomCreatives.java

```diff
@@ -0,0 +1,123 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dfp.axis.v201708.creativeservice;
+
+import com.beust.jcommander.Parameter;
+import com.google.api.ads.common.lib.auth.OfflineCredentials;
+import com.google.api.ads.common.lib.auth.OfflineCredentials.Api;
+import com.google.api.ads.common.lib.utils.Media;
+import com.google.api.ads.common.lib.utils.examples.CodeSampleParams;
+import com.google.api.ads.dfp.axis.factory.DfpServices;
+import com.google.api.ads.dfp.axis.v201708.Creative;
+import com.google.api.ads.dfp.axis.v201708.CreativeAsset;
+import com.google.api.ads.dfp.axis.v201708.CreativeServiceInterface;
+import com.google.api.ads.dfp.axis.v201708.CustomCreative;
+import com.google.api.ads.dfp.axis.v201708.CustomCreativeAsset;
+import com.google.api.ads.dfp.axis.v201708.Size;
+import com.google.api.ads.dfp.lib.client.DfpSession;
+import com.google.api.ads.dfp.lib.utils.examples.ArgumentNames;
+import com.google.api.client.auth.oauth2.Credential;
+import java.util.Random;
+
+/**
+ * This example creates a custom creative for a given advertiser. This feature
+ * is only available to DFP premium solution networks. To determine which
+ * creatives already exist, run GetAllCreatives.java.
+ *
+ * Credentials and properties in {@code fromFile()} are pulled from the
+ * "ads.properties" file. See README for more info.
+ */
+public class CreateCustomCreatives {
+
+  private static class CreateCustomCreativesParams extends CodeSampleParams {
+    @Parameter(names = ArgumentNames.ADVERTISER_ID, required = true,
+        description = "The ID of the advertiser (company) that all creatives will be assigned"
+            + " to.")
+    private Long advertiserId;
+  }
+
+  public static void runExample(DfpServices dfpServices, DfpSession session, long advertiserId)
+      throws Exception {
+    // Get the CreativeService.
+    CreativeServiceInterface creativeService =
+        dfpServices.get(session, CreativeServiceInterface.class);
+
+    // Create creative size.
+    Size size = new Size();
+    size.setWidth(600);
+    size.setHeight(315);
+    size.setIsAspectRatio(false);
+
+    // Create a custom creative.
+    CustomCreative customCreative = new CustomCreative();
+    customCreative.setName("Custom creative #" + new Random().nextInt(Integer.MAX_VALUE));
+    customCreative.setAdvertiserId(advertiserId);
+    customCreative.setDestinationUrl("http://google.com");
+    customCreative.setSize(size);
+
+    // Set the custom creative image asset.
+    CustomCreativeAsset customCreativeAsset = new CustomCreativeAsset();
+    customCreativeAsset.setMacroName("IMAGE_ASSET");
+    CreativeAsset asset = new CreativeAsset();
+    asset.setAssetByteArray(Media.getMediaDataFromUrl("https://goo.gl/3b9Wfh"));
+    // Filenames must be unique.
+    asset.setFileName(
+        String.format("image%s.jpg", new Random().nextInt(Integer.MAX_VALUE)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;nextInt&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 77. Consider formatting this data with &#39;%d&#39; instead.